src/util/include/util/TinyVector.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Xournal++
+ *
+ * A tiny vector stored in the stack, with a small maximal size
+ *
+ * @author Xournal++ Team
+ * https://github.com/xournalpp/xournalpp
+ *
+ * @license GNU GPLv2 or later
+ */
+
+#pragma once
+
+#include <array>
+#include <stdexcept>
+
+template <class T, unsigned int N>
+class TinyVector {
+public:
+    TinyVector() = default;
+
+    using size_type = unsigned int;
+    size_type size() const { return nb; }
+    bool empty() const { return nb == 0; }
+
+    T& operator[](size_type pos) { return dataArray[pos]; }
+    const T& operator[](size_type pos) const { return dataArray[pos]; }
+    T& front() { return dataArray.front(); }
+    const T& front() const { return dataArray.front(); }
+    T& back() { return dataArray[nb - 1]; }
+    const T& back() const { return dataArray[nb - 1]; }
+
+    using const_iterator = const T*;
+    using iterator = T*;
```

Comment:
You should use the array iterator here. It has more guarantees for debug compilation.

Reply:
Can I actually do both using `std::array::iterator` and using `std::aligned_storage`? If so, I don't know how.

Reply:
Right, that's not possible. You'll have to write your own then.